dataRetriever/getters/miniBlockDataGetter.go

```diff
@@ -0,0 +1,117 @@
+package getters
+
+import (
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("dataretriever/getters")
+
+// ArgMiniBlockDataGetter represents a miniblock data getter
+type ArgMiniBlockDataGetter struct {
+	MiniBlockPool    storage.Cacher
+	MiniBlockStorage storage.Storer
+	Marshalizer      marshal.Marshalizer
+}
+
+// miniBlockDataGetter is able to get data from cache or from storage
+type miniBlockDataGetter struct {
```

Comment:
maybe `Provider` instead of `Getter` ? also in package/directory name

Reply:
done